101-webapp-basic-windows/azuredeploy.json

```diff
@@ -1,58 +1,146 @@
+                }
+            },            
+            "node": {
+                "type": "object",
+                "defaultValue": {
+                    "appSettings": [
+                        {
+                            "name": "WEBSITE_NODE_DEFAULT_VERSION",
+                            "value": "12.15.0"
+                        }
+                    ]
+                },
+                "metadata": {
+                    "description": "Site config for Node.js."
+                }
+            },
+            "php": {
+                "type": "object",
+                "defaultValue": {
+                    "phpVersion": "7.4"
+                },
+                "metadata": {
+                    "description": "Site config for PHP."
+                }
+            }
+        }
     },
-    "location": {
-      "type": "string",
-      "defaultValue": "[resourceGroup().location]",
-      "metadata": {
-        "description": "Location for all resources"
-      }
-    }
-  },
-  "variables": {
-    "webAppPortalName": "[concat(parameters('webAppName'), '-webapp')]",
-    "appServicePlanName": "[concat('AppServicePlan-', parameters('webAppName'))]"
-  },
-  "resources": [
-    {
-      "apiVersion": "2018-02-01",
-      "type": "Microsoft.Web/serverfarms",
-      "kind": "app",
-      "name": "[variables('appServicePlanName')]",
-      "location": "[parameters('location')]",
-      "properties": {},
-      "dependsOn": [],
-      "sku": {
-        "name": "[parameters('sku')]"
-      }
-    },
-    {
-      "apiVersion": "2018-11-01",
-      "type": "Microsoft.Web/sites",
-      "kind": "app",
-      "name": "[variables('webAppPortalName')]",
-      "location": "[parameters('location')]",
-      "properties": {
-        "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', variables('appServicePlanName'))]"
-      },
-      "dependsOn": [
-        "[resourceId('Microsoft.Web/serverfarms', variables('appServicePlanName'))]"
-      ]
-    }
-  ]
-}
+    "resources": [
+        {
+            "apiVersion": "2019-08-01",
+            "type": "Microsoft.Web/serverfarms",
+            "name": "[variables('appServicePlanPortalName')]",
+            "location": "[parameters('location')]",
+            "sku": {
+                "name": "[parameters('sku')]"
+            }
+        },
+        {
+            "apiVersion": "2019-08-01",
+            "type": "Microsoft.Web/sites",
+            "kind": "app",
+            "name": "[parameters('webAppName')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Web/serverfarms', variables('appServicePlanPortalName'))]"
+            ],
+            "properties": {
+                "siteConfig": "[variables('configuration')]",
+                "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', variables('appServicePlanPortalName'))]"
+            },
+            "resources": [
```

Comment:
Why did we add this?  the original sample did not have an app deployed - and this changes the sample from an empty web app to one that's running a work load.

Also since the repo is not a param, the branch should not be either...

If we want to keep the repo option here we should provide a language option which is "none" and conditionally deploy the sourcecontrols resource.

Reverting to the original would also be fine if it fits with the doc.

Reply:
depending on the parameters used, yes the same webapp name will be generated - the uniqueString() function is idempotent.